iroha_p2p/src/peer.rs

```diff
@@ -183,10 +200,85 @@ where
         let read_half = self.read.as_mut().unwrap();
         let public_key = read_server_hello(read_half).await?;
         self.derive_shared_key(&public_key)?;
-        self.state = State::Ready;
+        self.state = State::SendKey;
+        Ok(())
+    }
+
+    /// Sends our app public key
+    async fn send_our_public_key(&mut self) -> Result<(), Error> {
+        debug!("Sending our public key...");
+        #[allow(clippy::unwrap_used)]
+        let write_half = self.write.as_mut().unwrap();
```

Comment:
When can write.as_mut fail?

Reply:
`write.as_mut()` fails only if we have `None` in `self.write`, but in these functions we already have the stream (its parts).

Reply:
Ok, I see. Then pleas add this to `.expect("Never fails as in in this function we already have the stream...")`